pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -134,25 +138,45 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 }
 
 func (pl *CSILimits) filterAttachableVolumes(
-	csiNode *storagev1.CSINode, volumes []v1.Volume, namespace string, result map[string]string) error {
-	for _, vol := range volumes {
-		// CSI volumes can only be used as persistent volumes
-		if vol.PersistentVolumeClaim == nil {
+	pod *v1.Pod, csiNode *storagev1.CSINode, newPod bool, result map[string]string) error {
+	for _, vol := range pod.Spec.Volumes {
+		// CSI volumes can only be used through a PVC.
+		pvcName := ""
+		ephemeral := false
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			if newPod && !pl.enableGenericEphemeralVolume {
+				return fmt.Errorf(
+					"volume %s is a generic ephemeral volume, but that feature is disabled in kube-scheduler",
+					vol.Name,
+				)
+			}
+			// Generic ephemeral inline volumes also use a PVC,
+			// just with a computed name, and...
+			pvcName = pod.Name + "-" + vol.Name
+			ephemeral = true
+		default:
 			continue
 		}
-		pvcName := vol.PersistentVolumeClaim.ClaimName
 
 		if pvcName == "" {
 			return fmt.Errorf("PersistentVolumeClaim had no name")
 		}
 
-		pvc, err := pl.pvcLister.PersistentVolumeClaims(namespace).Get(pvcName)
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
 
 		if err != nil {
-			klog.V(5).InfoS("Unable to look up PVC info", "PVC", fmt.Sprintf("%s/%s", namespace, pvcName))
+			klog.V(5).InfoS("Unable to look up PVC info", "PVC", fmt.Sprintf("%s/%s", pod.Namespace, pvcName))
```

Comment:
If we can't look up the PVC now, shouldn't we return unschedulable since we can't determine whether or not scheduling the pod on the node will exceed the volume limit?

Reply:
No idea. I don't know this code well enough to have an opinion about it.

Can we move this and the refactoring suggestion into issues and have someone work on that later? My goal with this KEP wasn't to change the underlying logic of how this filter works.

Reply:
sure, but I would like to get an answer if we should actually followup on this or not.

@msau42 can you answer this one?

Reply:
I think the concern here was for counting other pods that reference PVCs that don't exist. We don't want misconfigured other pods to block scheduling of the current pod. We probably could be more specific here and check for existence.

Reply:
We are using the same function for the incoming pod too when `newPod=true`. So should we return  unschedulable if this lookup fails and `newPod=true`?

Reply:
This question is not related to this PR, but rather the logic of this check without my modifications, right?

@msau42 : can you answer the question?

Reply:
Yes, if we can't find the PVCs for our own pod, then we should return an error. If it's another pod, we should ignore or handle "does not exist" specifically.

Agree we can handle this as a separate bug fix since this logic was already existing.

Reply:
thinking about this some more, @pohly does this become a bigger issue with ephemeral volumes since it's likely that a pvc may not exist immediately?

Reply:
I still don't understand all of the implications (= as in, what this function does and how it is called) to have an informed opinion.

From your older comment:
> Yes, if we can't find the PVCs for our own pod, then we should return an error. 

I've added that, regardless of `newPod`.

> If it's another pod, we should ignore or handle "does not exist" specifically.

With "it's another pod", do you mean the case where the PVC exists but is owned by another Pod? For the current Pod that's still the same problem: it needs a PVC and doesn't have it. Therefore I kept the "PVC is not owned by pod" error for that case.

Reply:
There were unit tests in `non_csi_test.go` that explicitly required that "missing PVC is not counted towards the PV limit". Did I understand correctly that you wanted me to change that so that missing PVCs are now errors?

I don't know what went into that original design decision.